netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -217,8 +217,8 @@ public Runnable start(Listener transportListener) {
      */
     b.handler(negotiationHandler);
     ChannelFuture regFuture = b.register();
-    channel = regFuture.channel();
-    if (channel == null) {
+    if (regFuture.isDone() && !regFuture.isSuccess()) {
```

Comment:
If channel can't be null, then it seems this if should be replaced with something that listens on `regFuture` like done for the writing of NOOP_MESSAGE.